tests/unit_tests/db_engine_specs/test_presto.py

```diff
@@ -106,3 +107,35 @@ def test_get_schema_from_engine_params() -> None:
         )
         is None
     )
+
+@mock.patch("superset.db_engine_specs.presto.PrestoEngineSpec.latest_partition")
+@pytest.mark.parametrize(
+    ["column_type", "column_value", "expected_value"],
+    [
+        (types.DATE(), "2023-05-01", "DATE '2023-05-01'"),
+        (types.TIMESTAMP(), "2023-05-01", "TIMESTAMP '2023-05-01'")
+    ]
```

Comment:
Actually further types, e.g. VARCHAR, isn't working for me here as the query still has a token in it in those cases, and can't be compared the way I'm doing it. I suspect if I can find the correct dialect I can first run it through `.compile(dialect=presto)` to fully compile that in, but in the current context I don't have a reference to the dialect and I don't see it under `sqlalchemy.dialects` so I'll have to find the correct dialect identifier or change the approach slightly.

Reply:
I've got this working with varchar and int types now.